src/js/spot/SpotItem.spec.jsx

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import userEvent from "@testing-library/user-event";
+import { build, fake } from "@jackfranklin/test-data-bot";
+import { screen, render as rtlRender } from "@testing-library/react";
+
+const buildSpotItem = build("SpotItem", {
```

Comment:
@boiarqin this is a bit of magic that I picked up from that Testing Javascript workshop. This allows us to focus on what's important during a test and let something else build out the data structure that we need.

Reply:
Very handy! I'm quite familiar the tools that it was inspired by. It's also nice that it has TS support.